### PR TITLE
LibGfx/TIFF: Don't stop decoding when failing to decode a tag

### PR DESCRIPTION
**Pull request by LucasChollet**
TIFF files are made in a way that make them easily extendable and over the years people have made sure to exploit that. In other words, it's easy to find images with non-standard tags. Instead of returning an error for that, let's skip them.

---

@nico

We would probably be able to decode the first layer of this image if we had support for CMYK.
[layers.tif.zip](https://github.com/SerenityOS/serenity/files/13796909/layers.tif.zip)

```
Setting image file directory pointer to 8
UnknownTag(254, UnsignedLong): 0
ImageWidth(UnsignedShort): 400
ImageHeight(UnsignedShort): 300
BitsPerSample(UnsignedShort): [8, 8, 8, 8]
Compression(UnsignedShort): NoCompression
PhotometricInterpretation(UnsignedShort): CMYK
StripOffsets(UnsignedLong): 73112
Orientation(UnsignedShort): Default
SamplesPerPixel(UnsignedShort): 4
RowsPerStrip(UnsignedShort): 300
StripByteCounts(UnsignedLong): 480000
XResolution(UnsignedRational): 300 (3000000/10000)
YResolution(UnsignedRational): 300 (3000000/10000)
PlanarConfiguration(UnsignedShort): Chunky
ResolutionUnit(UnsignedShort): Inch
UnknownTag(305, ASCII): Adobe Photoshop 25.1 (Macintosh)
UnknownTag(306, ASCII): 2023:12:29 21:38:32
Unable to decode tag 18/23
Unable to decode tag 19/23
Unable to decode tag 20/23
UnknownTag(700, Byte): Buffer of size: 14432
UnknownTag(34377, Byte): Buffer of size: 3032
UnknownTag(34665, UnsignedLong): 553112
Setting image file directory pointer to 493427
Runtime error: Unsupported value for PhotometricInterpretation
```

### CODE REVIEW

File: Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp [hunk 1]
```diff
@@ -360,8 +360,10 @@ class TIFFLoadingContext {
 
         auto const number_of_field = TRY(read_value<u16>());
 
-        for (u16 i = 0; i < number_of_field; ++i)
-            TRY(read_tag());
+        for (u16 i = 0; i < number_of_field; ++i) {
+            if (auto maybe_error = read_tag(); maybe_error.is_error() && TIFF_DEBUG)
```
Comment (nico) [inline Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp:364]: Maybe add a comment saying that we intentionally ignore tags that fail to decode.

**Review by nico — APPROVED** [review]
LG!

File: Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp (hunk 1 above)
Comment (LucasChollet) [reply Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp:364]: Do we really need that? When I read code, I consider the explicit `Error` management dance clear enough to know that the author intentionally did something special with the error. And in the worst case scenario, the commit message also supports the fact that this is very intentional.

Comment (nico) [reply Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp:364]: imho yes: There's no test protecting this from regressing, and `git blame` output becomes replaced through unrelated changes really quickly – try to look through `blame` output on an older file. It's usually 3-5 hops, often across a few renames, until you find the original commit message.

Comment (LucasChollet) [reply Userland/Libraries/LibGfx/ImageFormats/TIFFLoader.cpp:364]: Then we should write a test for protecting it from regressing, not write a comment!
And thanks for saying that, I discovered a bug by adding this test!

**Pull request merged**